src/include/common/copier_config/copier_config.h

```diff
@@ -31,22 +31,34 @@ struct CSVReaderConfig {
 };
 
 struct CopyDescription {
+    enum class CopyDirection : uint8_t { FROM = 0, TO = 1 };
+
     enum class FileType : uint8_t { UNKNOWN = 0, CSV = 1, PARQUET = 2, NPY = 3 };
 
-    CopyDescription(const std::vector<std::string>& filePaths, CSVReaderConfig csvReaderConfig,
-        FileType fileType);
+    // Copy From
+    CopyDescription(const std::vector<std::string>& copyFromFilePaths, FileType fileType,
+        CSVReaderConfig csvReaderConfig);
+
+    // Copy To
+    CopyDescription(const std::string& copyToFilePath, FileType fileType,
+        const std::vector<std::string>& columnNames);
 
     CopyDescription(const CopyDescription& copyDescription);
 
-    inline static std::string getFileTypeSuffix(FileType fileType) {
-        return "." + getFileTypeName(fileType);
-    }
+    inline static std::unordered_map<std::string, FileType> fileTypeMap{
+        {"unknown", FileType::UNKNOWN}, {".csv", FileType::CSV}, {".parquet", FileType::PARQUET},
+        {".npy", FileType::NPY}};
+
+    static FileType getFileTypeFromExtension(const std::string& extension);
 
     static std::string getFileTypeName(FileType fileType);
 
-    const std::vector<std::string> filePaths;
+    const std::vector<std::string> copyFromFilePaths;
+    const std::string copyToFilePath;
+    const std::vector<std::string> columnNames;
     std::unique_ptr<CSVReaderConfig> csvReaderConfig;
     FileType fileType;
+    CopyDirection copyDirection;
```

Comment:
I might be wrong, but seems you don't need to store copyDirection.

Also I would prefer to merge `copyFromFilePaths` and `copyToFilePath`. In the copyTo case, the vector should be of size 1.